### DISCUSSION TITLE
Help : not-supported:Cockpit is not compatible with the software on the system.

### DISCUSSION THREAD

**Discussion by zhaojing1987** [cockpit-project/cockpit#19484] [General]
ubuntu 22.04
cockpit 300.1

This is my code for test cockpit.http.request
```
cockpit.http({
            "address": "",
            "port": 8080,
        }).request({ path: "/apps/catalog", method: "GET" }).then((response) => {
            console.log("response:" + response);
        }).catch((error) => {
            console.log(error);
        });
```
This is the resust:
Object { problem: "not-supported", message: "Cockpit is not compatible with the software on the system.", toString: toString() }

**Comment by martinpitt** [comment]
This is not a valid way to call `.request()`, see [docs](https://cockpit-project.org/guide/latest/cockpit-http.html#cockpit-http-request). You need to either specify `body: ""` or call `.input()` first. But it's easier to do this:
```js
cockpit.http({ port: 8080 }).get("/apps/catalog")...
```
Still, this shouldn't cause a `not-supported` error, but just hang. So something else is wrong. Where/how exactly do you run this code?

**Comment by zhaojing1987** [comment]
I am customizing and developing a plugin for Cockpit, using create react app, which runs in React. Now that I have added body: '', it can run normally, but there is still something I don't understand. The interface I call through the request method is an API interface, and I need to handle different logic based on the status codes returned by the API, such as 200, 400, 404, and 500. However, it is how I should obtain the data returned by the API. It seems that I only obtained the data without the status code, But I tried using request. response ((status, headers)=>{...}), and here it seems that the body。
```
        cockpit.http({
            "address": "websoft9",
            "port": 8080,
        }).request({ path: “/apps/catalog”, method: "GET", body: "" }).then((response) => {
            console.log("status:" + response.status);
            console.log("body:" + response.body);
        }).catch((error) => {
             console.log(error);
        });
```
When the returned status code is 400, it is caught by the catch。 @martinpitt

**Comment by martinpitt** [comment]
Right, you'll get the `.then()` part for successful requests, and the `.catch()` part for failures like 4xx or 5xx. In that case `error.status` is the status code like 404:

```
>> cockpit.http({port: 8000}).request({ body: "", path: "/notexisting", method: "GET"}).catch(e => console.error(e))
Object { status: 404, reason: "File not found", message: "File not found", problem: null }
```

or alternatively if it's not a HTTP but a  connection error:
```
>> cockpit.http({port: 8001}).request({ body: "", path: "/", method: "GET"}).catch(e => console.error(e))
Object { problem: "not-found", message: "not-found", toString: toString() }
```
